backtrader/trade.py

```diff
@@ -68,6 +68,13 @@ def __init__(self,
         self.status.pnl = pnl
         self.status.pnlcomm = pnlcomm
         self.status.tz = tz
+        if event is not None:
+            self.event = event
+
+    def __reduce__(self):
+        return (self.__class__, (self.status.status, self.status.dt, self.status.barlen, self.status.size,
+                                 self.status.price, self.status.value, self.status.pnl, self.status.pnlcomm,
+                                 self.status.tz, self.event, ))
```

Comment:
Correct me if I'm wrong, but how will it work if the `event` argument to the `TradeHistory` ctor was not supplied - IIUC the `self.event` member will not be set in this case, no ?

Reply:
The event is updated during backtrader execution using doupdate method. The __init__ method never provides an event. This method will be called when unpickling and will call __init__ with all info from the object that was pickled.

Reply:
ok